### ISSUE TITLE
sql: back-to-back schema DROP DATABASE and CREATE DATABASE take up to 60s

### ISSUE THREAD

**Issue by otan** [cockroachdb/cockroach#46224]
**Describe the problem**

back-to-back drop database and create database queries can stall and take up to 60s.

**To Reproduce**

On one shell on the latest master: `./cockroach start-single-node --listen-addr=localhost:26257 --insecure`

On the next shell: `./cockroach sql --watch '1ms' -e 'DROP DATABASE IF EXISTS db0;CREATE DATABASE db0;USE db0;CREATE TABLE t0(c0 INT);' --insecure`

You can see 1 min query answers:
```

CREATE TABLE

Time: 46.067ms

CREATE TABLE

Time: 68.928ms

CREATE TABLE

Time: 43.123171s

CREATE TABLE

Time: 1m0.149811s

CREATE TABLE

Time: 1m0.026241s

CREATE TABLE

Time: 47.636ms

CREATE TABLE

Time: 47.969ms

CREATE TABLE

Time: 59.962868s
```

**Expected behavior**
Not taking a minute to resolve some queries

**Additional data / screenshots**

If additionally running with `./cockroach sql --watch '1ms' -e 'SHOW TABLES;' --insecure`, this will also spin for a while:

```

  table_name
---------------
  a
  geo_example
  nyc_streets
(3 rows)

Time: 56.238637s

  table_name
---------------
  a
  geo_example
  nyc_streets
(3 rows)

Time: 2.781ms
```

**Comment by otan** [comment]
cc @ajwerner / @lucy-zhang follow up from https://github.com/cockroachdb/cockroach/issues/24885#issuecomment-600321409

**Comment by awoods187** [comment]
Thanks for filing. I definitely think we should work on this in the near term.

**Comment by ajwerner** [comment]
The fact that this doesn't deadlock is a fun side-effect of #46170. In #46170 we now use high priority transactions when acquiring table leases. This acquisition can happen during re-planning of a query which has already laid down an intent on the `descriptor` or `namespace` table. Lease acquisitions also happens once a minute. #46170 didn't change the path where we resolve database descriptors (which have busted caching in terms of distributed state but do the right thing on the gateway which drops the database). The 60s behavior is really the periodic lease acquisition pushing the `DROP DATABASE` which deadlocked itself. 

The fix is to follow the pattern of #46170 on the database resolution path. PR inbound.
